module/userExtension/helper.js

```diff
@@ -0,0 +1,242 @@
+
+let sunBirdService =
+    require(ROOT_PATH + "/generics/services/sunbird");
+
+/**
+* user related information be here.
+* @method
+* @class  userExtensionHelper
+*/
+module.exports = class userExtensionHelper {
+
+
+    /**
+    * getProfile.
+    * @method
+    * @name  getProfile
+    * @param  {queryObject, token}  - requested queryObject and token.
+    * @returns {json} Response consists of user profile details.
+    */
+    static getProfile(queryObject, token) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let profileData = await sunBirdService.getUserProfileInfo(token, queryObject.userId);
+                let roles;
+                profileData = JSON.parse(profileData);
+                if (profileData.responseCode == messageConstants.common.RESPONSE_OK) {
+                    if (profileData.result &&
+                        profileData.result.response
+                    ) {
+
+                        if (profileData.result.response.roles) {
+                            roles = profileData.result.response.roles;
+                        }
+                        await Promise.all(profileData.result.response.organisations.map(orgInfo => {
+                            if (orgInfo.roles) {
+                                roles.push(...orgInfo.roles);
+                            }
+                        }))
+                    }
+                }
+
+                if (roles) {
+                    roles = Array.from(new Set(roles));
+                }
+
+                let userExtDocument = await database.models.userExtension.findOne(
+                    queryObject,
+                    {
+                        updatedBy: 0,
+                        createdBy: 0,
+                        createdAt: 0,
+                        updatedAt: 0,
+                        status: 0,
+                        deleted: 0,
+                        userId: 0,
+                        "__v": 0
+                    }
+                ).lean();
+
+                if (userExtDocument) {
+                    userExtDocument.roles = userExtDocument.roles.map(role => role.code)
+                    if (roles) {
+                        userExtDocument.roles.push(...roles);
+                    }
+                }
+                return resolve(userExtDocument);
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+
+    }
+
+
+    /**
+ * create.
+ * @method
+ * @name  create
+ * @param  {request, token, userId}  - requested body, token and userId.
+ * @returns {json} Response consists of created user details.
+ */
+    static create(request, token, userId) {
+
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (request.organisation &&
```

Comment:
@rakeshSgr , @deepakr123 - All the checks can be made in request validator before the controller is called.